api/v1alpha1/search_types.go

```diff
@@ -0,0 +1,199 @@
+	// Configuration for collector Deployment
+	Collector DeploymentConfig `json:"collector,omitempty"`
+
+	// +optional
+	// Configuration for api Deployment
+	API DeploymentConfig `json:"api,omitempty"`
+
+	// +optional
+	// Configuration for addon installed collector Deployment
+	RemoteCollector DeploymentConfig `json:"remoteCollector,omitempty"`
+}
+
+type DeploymentConfig struct {
+	// +optional
+	// Number of pod instances for deployment
+	ReplicaCount int `json:"replicaCount,omitempty"`
+	// +optional
+	// Compute Resources required by deployment
+	Resources *corev1.ResourceRequirements `json:"resources,omitempty"`
+	//Image_override
+	ImageOverride string `json:"imageOverride,omitempty"`
+
+	// +optional
+	//ImagePullSecret
+	ImagePullSecret string `json:"imagePullSecret,omitempty"`
+
+	//ImagePullPolicy
+	// +optional
+	ImagePullPolicy corev1.PullPolicy `json:"imagePullPolicy,omitempty"`
+
+	// NodeSelector to schedule on nodes with matching labels
+	// +optional
+	NodeSelector map[string]string `json:"nodeSelector,omitempty"`
+
+	//Proxy config , if remote collectors need to override
+	// +optional
+	ProxyConfig map[string]string `json:"proxyConfig,omitempty"`
+}
+
+type StorageSpec struct {
+	// +optional
+	// name of the storage class
+	StorageClassName string `json:"storageClassName,omitempty"`
+	// +optional
+	// storage capacity
+	Size *resource.Quantity `json:"size,omitempty"`
+}
+
+type FilterSpec struct {
+	// +optional
+	// Allowed resources from collector
+	AllowedResources ResourceListSpec `json:"allowedResources,omitempty"`
+
+	// +optional
+	// Denied resources from collector
+	DeniedResources ResourceListSpec `json:"deniedResources,omitempty"`
+}
+
+type ResourceListSpec struct {
+	//API Group names to be filtered
+	APIGroups []string `json:"apiGroups,omitempty"`
+	//Resource names to be filtered
+	Resources []string `json:"resources,omitempty"`
+	// +optional
+	//Cluster Labels this filter to be applied
+	ClusterLabels []string `json:"clusterLabels,omitempty"`
+}
+
+// SearchStatus defines the observed state of Search
+type SearchStatus struct {
+	// INSERT ADDITIONAL STATUS FIELD - define observed state of cluster
+	// Important: Run "make" to regenerate code after modifying this file
+	// Human readable health state
+	SearchHealth string `json:"searchHealth,omitempty"`
+	// Database used by search
+	DBInUse string `json:"dbInUse,omitempty"`
+	// Storage used by database
+	StorageInUse string `json:"storageInUse,omitempty"`
+	// +optional
+	Conditions SearchConditions `json:"conditions,omitempty"`
+}
+
+type SearchCondition struct {
+	Type   SearchConditionType    `json:"type"`
+	Status corev1.ConditionStatus `json:"status"`
+	// Last time the condition transitioned
+	LastTransitionTime metav1.Time `json:"lastTransitionTime"`
+	// +optional
+	// Reason for the condition's last transition.
+	Reason string `json:"reason,omitempty"`
+	// +optional
+	// Human readable message
+	Message string `json:"message,omitempty" `
+}
```

Comment:
Should use the generic `Condition` from the Kubernetes API? https://pkg.go.dev/k8s.io/apimachinery/pkg/apis/meta/v1#Condition

Reply:
What's the concern with upgrade?  If the kube API changes it will need to be published as `v2`, so it won't break us.

My concern is consistency. This is a concept that kubernetes has standardized across resources, so I don't want us to drift.